misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java

```diff
@@ -236,6 +240,19 @@ private JsonObject visibleEP(Node node, TypeSymbol typeSymbol) {
                 symbolMetaInfo.addProperty("orgName", orgName);
                 symbolMetaInfo.addProperty("moduleName", moduleName);
             }
+        } else if (node.kind() == SyntaxKind.MODULE_VAR_DECL) {
+            JsonObject metaInfoForModuleVar = new JsonObject();
+            ModuleVariableDeclarationNode variableDeclarationNode = (ModuleVariableDeclarationNode) node;
+            CaptureBindingPatternNode captureBindingPatternNode =
+                    (CaptureBindingPatternNode) variableDeclarationNode.typedBindingPattern().bindingPattern();
+            metaInfoForModuleVar.addProperty("name", captureBindingPatternNode.variableName().text());
+            metaInfoForModuleVar.addProperty("isCaller", "Caller".equals(typeSymbol.getName()
```

Comment:
Type name cannot always be caller. Hence we have to check the semantic properties to verify whether it is a caller or not. Otherwise this is going to be a specific case being handled.

Reply:
it is a specific case being handled here.... This is very specific to HTTP:Caller

Reply:
Since this mapper is a public API and can be used by other parties as well, shall we use the semantic API to get the client information? Also, in language point of view, are going to refer a client?

Reply:
this is very specific to the diagram ... this Json generator is basically built for that... and what API we can use to get the client information? ... this is very very specific to identifying caller in HTTP resource.